container_pipeline/workers/build.py

```diff
@@ -0,0 +1,76 @@
+#!/usr/bin/env python
+import hashlib
+import json
+import logging
+import os
+
+from container_pipeline.utils import Build, get_job_name
+from container_pipeline.lib.logging import load_logger
+from .base import BaseWorker
+
+
+class BuildWorker(BaseWorker):
+    """Build worker"""
+
+    def handle_job(self, job):
+        """Handle job"""
+        parent_build_running = False
+        parents = job.get('depends_on', '').split(',')
+        parents_in_build = []
+        for parent in parents:
+            is_build_running = Build(parent).is_running()
+            if is_build_running:
+                parents_in_build.append(parent)
+            parent_build_running = parent_build_running or \
+                is_build_running
+
+        if parent_build_running:
+            self.logger.info('Parents in build: %s, pushing job: %s back '
+                             'to queue' % (parents_in_build, job))
+            self.queue.put(job, 'master_tube')
+        else:
+            self.logger.info('Starting build for job: %s' % job)
+            success = self.build(job)
+            if success:
+                self.handle_build_success(job)
+            else:
+                self.handle_build_failure(job)
+
+    def build(self, job):
+        """Build job"""
+        namespace = get_job_name(job)
+        project = hashlib.sha224(namespace).hexdigest()
+
+        build_id = self.openshift.build(project, 'build')
+        Build(namespace).start()
+        build_status = self.openshift.wait_for_build_status(
+            project, build_id, 'Complete')
+        logs = self.openshift.get_build_logs(project, build_id)
+        build_logs_file = os.path.join(job['logs_dir'], 'build_logs.txt')
+        self.export_build_logs(logs, build_logs_file)
+        return build_status
+
+    def handle_build_success(self, job):
+        """Handle build success for job"""
+        self.logger.debug("Build is successful going for next job")
+
+    def handle_build_failure(self, job):
+        """Handle build failure for job"""
+        self.queue.put(json.dumps(job), 'failed_build')
+        self.logger.info(
+            "Build is not successful putting it to failed build tube")
+        appid = job['appid']
+        jobid = job['jobid']
+        desired_tag = job['desired_tag']
+        project = appid + "/" + jobid + ":" + desired_tag
```

Comment:
Ack! This validation needs to happen at an earlier stage. Few stages I can imagine are:
- when someone opens a PR on container-index, index CI should check it. cc: @mohammedzee1000 
- when Jenkins puts the data on to the tube, we should be sure that it won't put erroneous/incomplete data. This one has particularly bit us in past when `TEST_TAG` was missing and mail server went into a frenzy.